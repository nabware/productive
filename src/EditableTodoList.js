import React from "react";
import EditableTodo from "./EditableTodo";

/** Show list of editable todos.
 *
 * Props:
 * - todos: array of [ todo, ... ]
 * - update(): fn to call to update a todo
 * - remove(): fn to call to remove a todo
 *
 * TodoApp -> EditableTodoList -> [ EditableTodo, ... ]
 */

function EditableTodoList({todos}) {
  return (
      <div>
        {todos.map(t => <EditableTodo todo={t} /> )}
      </div>
  );
}

export default EditableTodoList;
