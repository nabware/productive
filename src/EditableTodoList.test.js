import React from "react";
import { render, fireEvent } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList.js";
import TEST_TODOS from "./_testCommon.js";

describe("smoketest", function () {
  it("renders without crashing", function () {
    render(<EditableTodoList todos={TEST_TODOS} />);
  });

  it("displays all todos", function () {
    const { getByText } = render(<EditableTodoList todos={TEST_TODOS} />);

    expect(getByText("test1title")).toBeInTheDocument();
    expect(getByText("test1description")).toBeInTheDocument();
    expect(getByText("(priority: 1)")).toBeInTheDocument();

    expect(getByText("test2title")).toBeInTheDocument();
    expect(getByText("test2description")).toBeInTheDocument();
    expect(getByText("(priority: 2)")).toBeInTheDocument();

    expect(getByText("test3title")).toBeInTheDocument();
    expect(getByText("test3description")).toBeInTheDocument();
    expect(getByText("(priority: 3)")).toBeInTheDocument();
  });

  it("snapshot", function () {
    const { container } = render(<EditableTodoList todos={TEST_TODOS} />);
    expect(container).toMatchSnapshot();
  });
});

