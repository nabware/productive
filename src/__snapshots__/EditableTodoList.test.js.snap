// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoketest snapshot 1`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              test1title
            </b>
             
            <small>
              (priority: 
              1
              )
            </small>
          </div>
          <div>
            <small>
              test1description
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              test2title
            </b>
             
            <small>
              (priority: 
              2
              )
            </small>
          </div>
          <div>
            <small>
              test2description
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              test3title
            </b>
             
            <small>
              (priority: 
              3
              )
            </small>
          </div>
          <div>
            <small>
              test3description
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`todo form snapshot 1`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              test1title
            </b>
             
            <small>
              (priority: 
              1
              )
            </small>
          </div>
          <div>
            <small>
              test1description
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              test2title
            </b>
             
            <small>
              (priority: 
              2
              )
            </small>
          </div>
          <div>
            <small>
              test2description
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              test3title
            </b>
             
            <small>
              (priority: 
              3
              )
            </small>
          </div>
          <div>
            <small>
              test3description
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
